polyply/tests/test_lib_files.py

```diff
@@ -0,0 +1,244 @@
+                print(inter)
+                print(new_interactions)
+                print("--------------------------------------")
+                assert sum([_interaction_equal(inter, inter_new, inter_type) for inter_new in new_interactions[:len(ref_interactions)]]) == 1
+
+
+def compare_itp(filename1, filename2):
+    """
+    Asserts that two itps are functionally identical
+    """
+    dummy_ff = ForceField(name='dummy')
+    with open(filename1) as fn1:
+        vermouth.gmx.read_itp(fn1, dummy_ff)
+    dummy_ff2 = ForceField(name='dummy')
+    with open(filename2) as fn2:
+        vermouth.gmx.read_itp(fn2, dummy_ff2)
+    for block in dummy_ff2.blocks.values():
+        block._force_field = dummy_ff
+    assert set(dummy_ff.blocks.keys()) == set(dummy_ff2.blocks.keys())
+    for name in dummy_ff.blocks:
+        block1 = dummy_ff.blocks[name]
+        block2 = dummy_ff2.blocks[name]
+        assert_equal_blocks(block1, block2)
+
+
+def compare_pdb(filename1, filename2):
+    """
+    Asserts that two pdbs are functionally identical
+    """
+    pdb1 = vermouth.pdb.read_pdb(filename1)
+    pdb2 = vermouth.pdb.read_pdb(filename2)
+    assert len(pdb1) == len(pdb2)
+    for mol1, mol2 in zip(pdb1, pdb2):
+        for mol in (mol1, mol2):
+            for n_idx in mol:
+                node = mol.nodes[n_idx]
+                if 'position' in node and node['atomname'] in ('SCN', 'SCP'):
+                    # Charge dummies get placed randomly, which complicated
+                    # comparisons to no end.
+                    # These will be caught by the distances in the edges instead.
+                    del node['position']
+
+        assert_equal_blocks(mol1, mol2)
+
+
+COMPARERS = {'.itp': compare_itp,
+             '.pdb': compare_pdb}
+
+
+def _interaction_equal(interaction1, interaction2, inter_type):
+    """
+    Returns True if interaction1 == interaction2, ignoring rounding errors in
+    interaction parameters.
+    """
+    p1 = list(map(str, interaction1.parameters))
+    p2 = list(map(str, interaction2.parameters))
+    a1 = list(interaction1.atoms)
+    a2 = list(interaction2.atoms)
+    if inter_type in ["constraints", "bonds", "exclusions", "pairs"]:
+        a1.sort()
+        a2.sort()
+        return a1 == a2 and p1 == p2
+
+    elif inter_type in ["impropers", "dihedrals"]:
+         if frozenset([a1[0], a1[1], a1[2]]) == frozenset([a2[0], a2[1], a2[2]]) and a1[3] == a2[3] and p1 == p2:
+            return True
+         elif frozenset([a1[0], a1[1], a1[2]]) == frozenset([a2[3], a2[2], a2[1]]) and a1[3] == a2[0] and p1 == p2:
+            return True
+         elif frozenset([a1[1], a1[2], a1[3]]) == frozenset([a2[1], a2[2], a2[3]]) and a1[0] == a2[0] and p1 == p2:
+            return True
+
+    elif inter_type in ["angles"]:
+        return a1[1] == a2[1] and frozenset([a1[0], a1[2]]) == frozenset([a2[0], a2[2]]) and p1 == p2
+
+    return False
```

Comment:
Add an `if p1 != p2: return False` early on to simplify the rest of the logic a little